common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -57,11 +59,40 @@ public static void enableOverlays(List<Overlay> overlays, boolean ignoreState) {
 
     @SubscribeEvent
     public static void onRenderPre(RenderEvent.Pre event) {
+        assertSectionsAreCorrect();
+
         McUtils.mc().getProfiler().push("preRenOverlay");
         renderOverlays(event, OverlayInfo.RenderState.Pre);
         McUtils.mc().getProfiler().pop();
     }
 
+    private static void assertSectionsAreCorrect() {
+        if (sections.isEmpty()) {
+            WynntilsMod.error("We had no calculated sections when trying to render! Forcing recalculation.");
+            calculateSections();
+            return;
+        }
+
+        // this assumes sections are equal in size
+        Window window = McUtils.window();
+        SectionCoordinates section = sections.get(0);
+
+        // account for loss of pixels when section does not perfectly divide screen
+        int sqrt = (int) Math.sqrt(OVERLAY_SECTIONS);
```

Comment:
Why do we use square roots?

Reply:
9 sections = 3 sections per row or  per column

Reply:
I think that just makes it more complicated and can be simplified.

Reply:
I mean it was like this before

Reply:
I'd rather we go
```java
private static final int SECTION_SIDE_NUMBER = 3;

private static final List<SectionCoordinates> sections = new ArrayList<>(SECTION_SIDE_NUMBER * SECTION_SIDE_NUMBER);
```